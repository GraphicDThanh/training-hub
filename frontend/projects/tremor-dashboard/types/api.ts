/**
 * If you want the API to return something for your ease of handling the UI, style that variable here
 */
export interface IResponseFromAPI<Q> {
  errorMessages: Array<Record<string, string>>;
  data: Q | null;
}
