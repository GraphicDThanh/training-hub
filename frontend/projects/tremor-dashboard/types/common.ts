export interface Step {
  index: number;
  content: string;
}

export interface IMedia {
  image: string;
}

export interface CdnResponse {
  data: {
    width: number;
    height: number;
    image: ImageInfo;
  };
}

export interface ImageInfo {
  filename: string;
  url: string;
}

export interface SortItem {
  key: string;
  direction: string;
}

export type TSearchParams = {
  id?: string;
  query?: string;
  isAvailable?: string;
  status?: string;
  page?: number;
  sortBy?: string;
  orderBy?: string;
};

export interface TFieldSocial {
  name: string;
  id: string;
  label: string;
  value: string;
}
