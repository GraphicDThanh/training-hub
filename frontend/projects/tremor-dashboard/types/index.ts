export * from "./analytics";
export * from "./api";
export * from "./billing";
export * from "./breadcrumb";
export * from "./common";
export * from "./invoice";
export * from "./navigation";
export * from "./orderDetails";
export * from "./orders";
export * from "./product";
export * from "./profile";
export * from "./project";
export * from "./sales";
export * from "./select";
export * from "./table-list";
export * from "./table";
export * from "./transactions";
export * from "./user";
