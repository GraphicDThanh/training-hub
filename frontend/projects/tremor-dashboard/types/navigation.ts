export interface INavigation {
  icon?: JSX.Element;
  label: string;
  href: string;
}
