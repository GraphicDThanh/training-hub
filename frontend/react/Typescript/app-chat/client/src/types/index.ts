export type User = {
  id: string;
  name: string;
  avatar?: string;
  isCurrent?: boolean;
}

export type Message = {
  id: string;
  author: User;
  content: string;
}

export type StoreState = {
  messages: Message[];
  users: User[];
  loading: string[];
  error: object;
  modal: any;
  auth: any;
};

export type Action = {
  type: string;
}

export type Account = {
  email: string;
  password: string;
}