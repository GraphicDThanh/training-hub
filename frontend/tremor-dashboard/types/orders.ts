export interface OrderProduct {
  id: number;
  name: string;
  count: number;
  price?: number;
  url?: string;
}

export interface Order {
  id: number;
  createdAt: string;
  status: number;
  customer: {
    id: number;
    fullName: string;
    avatar?: string;
  };
  products: OrderProduct[];
  revenue: number;
}

export interface OrderResponse {
  results: Order[];
  total: number;
  message?: string;
}
