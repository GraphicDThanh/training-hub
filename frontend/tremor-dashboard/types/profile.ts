import { USER_ROLE } from "./user";

export interface Conversation {
  id: string;
  message: string;
  createdAt: string;
  userId: number;
}

export interface ConversationGroup {
  id: string;
  lastConversation?: string;
  userIds: number[];
}

export interface PlatformSetting {
  label: string;
  field: string;
}

export interface AccountSettingData {
  [key: string]: boolean;
  emailMentions: boolean;
  emailFollowing: boolean;
  emailAnswerPost: boolean;
}

export interface ApplicationSettingData {
  [key: string]: boolean;
  newLaunchesProject: boolean;
  monthlyProductUpdate: boolean;
  subscribeToNewsletter: boolean;
}

export interface SocialLink {
  facebook?: string;
  twitter?: string;
  instagram?: string;
}

export interface UserProfile {
  id: number;
  created_at: string;
  name: string;
  role: USER_ROLE;
  information: string;
  account_setting: AccountSettingData;
  application_setting: ApplicationSettingData;
  phone: string;
  email: string;
  location: string;
  conversations: Conversation[];
  socials: SocialLink;
  avatar: string;
}

export interface Message {
  message: string;
}

export interface ChatBox extends Message {
  userId: number;
  messageId: string;
  time: string;
  avatar: string;
}
