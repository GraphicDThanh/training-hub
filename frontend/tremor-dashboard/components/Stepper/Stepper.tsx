// Components
import { Text } from "@tremor/react";

// Types
import { Step } from "@/types";

// Styles
import "@/styles/stepper.css";

interface StepperProps {
  currentStep: number;
  steps: Step[];
  total: number;
  extendClass?: string;
}

const Stepper = ({ currentStep, steps, total, extendClass }: StepperProps) => {
  const isFirstStep = currentStep === 1;
  const isLastStep = currentStep === total;
  const firstStep = steps[0];
  const lastStep = steps[total - 1];
  const bodySteps = steps.slice(1, total - 1);

  // Fist Item class
  const firstItemRightLineClass = `1234 h-0.5 w-full ${
    isFirstStep ? "bg-zinc-400" : "bg-white"
  }`;

  // Last Item class
  const lastItemLeftLineClass = `h-0.5 w-[50%] ${
    isLastStep ? "bg-white" : "bg-zinc-400"
  }`;
  const lastItemDotClass = isLastStep ? "stepper-dot-active" : "stepper-dot";
  const lastItemTextClass = `stepper-text-content mt-2 ${
    isLastStep ? "text-white" : "text-silver dark:silver"
  }`;

  // Body Item
  const bodyItems = bodySteps.map(step => {
    const stepIndex = step.index;
    const leftLineClass = `h-0.5 w-[50%] ${
      currentStep >= stepIndex ? "bg-white" : "bg-zinc-400"
    }`;
    const dotClass =
      currentStep >= stepIndex ? "stepper-dot-active" : "stepper-dot";
    const rightLineClass = `h-0.5 w-[50%] ${
      currentStep > stepIndex ? "bg-white" : "bg-zinc-400"
    }`;
    const textClass = `stepper-text-content mt-2 ${
      currentStep >= stepIndex ? "text-white" : "text-silver dark:text-silver"
    }`;

    return (
      <li key={step.content} data-testid={`step-content-${step.index}`}>
        <div className="flex items-center w-full h-4">
          {/* Left line */}
          <span className={leftLineClass}></span>
          {/* Dot */}
          <span className={dotClass}></span>
          {/* Right line */}
          <span className={rightLineClass}></span>
        </div>
        <Text
          className={`${textClass} text-xs`}>{`${step.index}. ${step.content}`}</Text>
      </li>
    );
  });

  return (
    <ul
      data-testid="stepper"
      className={`grid grid-cols-${total} pt-6 pb-4 bg-zinc-700 dark:stepper-dark-bg shadow-box-icon-primary rounded-lg absolute -translate-y-10 z-10 w-11/12 dark:bg-gradient-pickled ${extendClass}`}>
      {/* First Item */}
      <li data-testid="step-content-0">
        <div className="flex items-center w-full h-4 pl-[50%]">
          <span className="stepper-dot-active"></span>
          <span className={firstItemRightLineClass}></span>
        </div>
        <Text className="stepper-text-content text-xs text-white mt-2">{`${firstStep.index}. ${firstStep.content}`}</Text>
      </li>
      {/* Body Items */}
      {bodyItems}
      {/* Last Item */}
      <li data-testid="last-step-content">
        <div className="flex items-center w-full h-4">
          <span className={lastItemLeftLineClass}></span>
          <span className={lastItemDotClass}></span>
        </div>
        <p
          className={
            lastItemTextClass
          }>{`${lastStep.index}. ${lastStep.content}`}</p>
      </li>
    </ul>
  );
};

export default Stepper;
