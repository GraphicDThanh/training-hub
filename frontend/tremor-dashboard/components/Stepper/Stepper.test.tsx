// Libs
import { render } from "@testing-library/react";

// Components
import Stepper from "./Stepper";

// Types
import { Step } from "@/types";

describe("Stepper component testing", () => {
  const mockSteps: Step[] = [
    { index: 1, content: "Product Info" },
    { index: 2, content: "Media" },
    { index: 3, content: "Social" },
    { index: 4, content: "Pricing" },
  ];
  it("should match snapshot first step", () => {
    const { container } = render(
      <Stepper steps={mockSteps} currentStep={1} total={mockSteps.length} />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot last step", () => {
    const { container } = render(
      <Stepper steps={mockSteps} currentStep={4} total={mockSteps.length} />,
    );

    expect(container).toMatchSnapshot();
  });
});
