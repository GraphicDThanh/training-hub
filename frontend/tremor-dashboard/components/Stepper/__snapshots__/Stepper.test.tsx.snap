// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper component testing should match snapshot first step 1`] = `
<div>
  <ul
    class="grid grid-cols-4 pt-6 pb-4 bg-zinc-700 dark:stepper-dark-bg shadow-box-icon-primary rounded-lg absolute -translate-y-10 z-10 w-11/12 dark:bg-gradient-pickled undefined"
    data-testid="stepper"
  >
    <li
      data-testid="step-content-0"
    >
      <div
        class="flex items-center w-full h-4 pl-[50%]"
      >
        <span
          class="stepper-dot-active"
        />
        <span
          class="1234 h-0.5 w-full bg-zinc-400"
        />
      </div>
      <p
        class="text-tremor-default dark:text-dark-tremor-content stepper-text-content text-xs text-white mt-2"
      >
        1. Product Info
      </p>
    </li>
    <li
      data-testid="step-content-2"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-zinc-400"
        />
        <span
          class="stepper-dot"
        />
        <span
          class="h-0.5 w-[50%] bg-zinc-400"
        />
      </div>
      <p
        class="text-tremor-default stepper-text-content mt-2 text-silver dark:text-silver text-xs"
      >
        2. Media
      </p>
    </li>
    <li
      data-testid="step-content-3"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-zinc-400"
        />
        <span
          class="stepper-dot"
        />
        <span
          class="h-0.5 w-[50%] bg-zinc-400"
        />
      </div>
      <p
        class="text-tremor-default stepper-text-content mt-2 text-silver dark:text-silver text-xs"
      >
        3. Social
      </p>
    </li>
    <li
      data-testid="last-step-content"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-zinc-400"
        />
        <span
          class="stepper-dot"
        />
      </div>
      <p
        class="stepper-text-content mt-2 text-silver dark:silver"
      >
        4. Pricing
      </p>
    </li>
  </ul>
</div>
`;

exports[`Stepper component testing should match snapshot last step 1`] = `
<div>
  <ul
    class="grid grid-cols-4 pt-6 pb-4 bg-zinc-700 dark:stepper-dark-bg shadow-box-icon-primary rounded-lg absolute -translate-y-10 z-10 w-11/12 dark:bg-gradient-pickled undefined"
    data-testid="stepper"
  >
    <li
      data-testid="step-content-0"
    >
      <div
        class="flex items-center w-full h-4 pl-[50%]"
      >
        <span
          class="stepper-dot-active"
        />
        <span
          class="1234 h-0.5 w-full bg-white"
        />
      </div>
      <p
        class="text-tremor-default dark:text-dark-tremor-content stepper-text-content text-xs text-white mt-2"
      >
        1. Product Info
      </p>
    </li>
    <li
      data-testid="step-content-2"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-white"
        />
        <span
          class="stepper-dot-active"
        />
        <span
          class="h-0.5 w-[50%] bg-white"
        />
      </div>
      <p
        class="text-tremor-default dark:text-dark-tremor-content stepper-text-content mt-2 text-white text-xs"
      >
        2. Media
      </p>
    </li>
    <li
      data-testid="step-content-3"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-white"
        />
        <span
          class="stepper-dot-active"
        />
        <span
          class="h-0.5 w-[50%] bg-white"
        />
      </div>
      <p
        class="text-tremor-default dark:text-dark-tremor-content stepper-text-content mt-2 text-white text-xs"
      >
        3. Social
      </p>
    </li>
    <li
      data-testid="last-step-content"
    >
      <div
        class="flex items-center w-full h-4"
      >
        <span
          class="h-0.5 w-[50%] bg-white"
        />
        <span
          class="stepper-dot-active"
        />
      </div>
      <p
        class="stepper-text-content mt-2 text-white"
      >
        4. Pricing
      </p>
    </li>
  </ul>
</div>
`;
